src/api/v1beta1/properties.go

```diff
@@ -257,15 +262,19 @@ func tenantUUID(apiUrl string) (string, error) {
 		return "", errors.WithMessagef(err, "problem parsing tenant id from url %s", apiUrl)
 	}
 
-	fqdn := parsedUrl.Hostname()
-	hostnameWithDomains := strings.FieldsFunc(fqdn,
-		func(r rune) bool { return r == '.' },
-	)
-	if len(hostnameWithDomains) < 1 {
-		return "", fmt.Errorf("problem getting tenant id from fqdn '%s'", fqdn)
+	// Path = "/e/<token>/api" -> ["e",  "<tenant>", "api"]
+	subPaths := util.Tokenize(parsedUrl.Path, '/')
```

Comment:
is there an advantage to using this instead of `strings.Split()`?

Reply:
It's based on `strings.FieldsFunc()` which combines adjacent separators and skips empty split results automatically (in particular leading and/or trailing), [see playground example](https://go.dev/play/p/10LFof1C1ql).

`FieldsFunc` have an overcomplicated API which is why I wrapped it in `Tokenize`.

Reply:
I think we can use the STL for this: [url.Parse](https://pkg.go.dev/net/url#Parse)

Reply:
`url.Parse()` serves different purpose - extracting protocol, hostname & domains, path & query arguments. It doesn't help with splitting the path into tokens.

I thought of [`path.Split()` ](https://pkg.go.dev/net/path#Split) as well and it doesn't do what I need. Either way, a helper function is needed, be it local or exported, named `Tokenize()` or something similar.

Reply:
I just don't see the advantage of a split function that ignores empty results (unless beginning/end). That sounds like a malformed URL to me that shouldn't be parsed without error.
`path.Split()` is for file paths, so I don't think this should be used here either.

Reply:
It's how STL `strings.FieldsFunc()` works - it was implemented and accepted this way before. Removed the `Tokenize()` wrapper to not fuel further discussion.